app/src/main/java/io/kuenzler/whatsappwebtogo/BlobConverter.java

```diff
@@ -0,0 +1,144 @@
+                extension = ".avi";
+                break;
+            case "data:image/gif;base64":
+                extension = ".gif";
+                break;
+            case "data:application/pdf;base64":
+                extension = ".pdf";
+                break;
+            default://should write cases for more images types
+                extension = "jpg";
+                break;
+        }
+
+        final File dwldsPath = new File(Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOWNLOADS)+" "+currentDateTime+"downloaded" + extension);
+        byte[] pdfAsBytes = Base64.decode(base64PDf.replaceFirst(strings[0], ""), 0);
+        FileOutputStream os;
+        os = new FileOutputStream(dwldsPath, false);
+        os.write(pdfAsBytes);
+        os.flush();
+
+        if (dwldsPath.exists()) {
+            Intent intent = new Intent();
+            intent.setAction(Intent.ACTION_VIEW);
+            Uri apkURI = FileProvider.getUriForFile(context,context.getApplicationContext().getPackageName() + ".provider", dwldsPath);
+            intent.setDataAndType(apkURI, MimeTypeMap.getSingleton().getMimeTypeFromExtension(extension));
+            intent.addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
+            PendingIntent pendingIntent = PendingIntent.getActivity(context,1, intent, PendingIntent.FLAG_CANCEL_CURRENT);
+            String CHANNEL_ID = "MYCHANNEL";
+            final NotificationManager notificationManager = (NotificationManager) context.getSystemService(Context.NOTIFICATION_SERVICE);
+
+            if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.O) {
+                NotificationChannel notificationChannel= new NotificationChannel(CHANNEL_ID,"name", NotificationManager.IMPORTANCE_LOW);
+                Notification notification = new Notification.Builder(context,CHANNEL_ID)
+                        .setContentText("You have got something new!")
+                        .setContentTitle("File downloaded")
+                        .setContentIntent(pendingIntent)
+                        .setChannelId(CHANNEL_ID)
+                        .setSmallIcon(android.R.drawable.sym_action_chat)
+                        .build();
+                if (notificationManager != null) {
+                    notificationManager.createNotificationChannel(notificationChannel);
+                    notificationManager.notify(notificationId, notification);
+                }
+
+            } else {
+                NotificationCompat.Builder b = new NotificationCompat.Builder(context, CHANNEL_ID)
+                        .setDefaults(NotificationCompat.DEFAULT_ALL)
+                        .setWhen(System.currentTimeMillis())
+                        .setSmallIcon(android.R.drawable.sym_action_chat)
+                        //.setContentIntent(pendingIntent)
+                        .setContentTitle("Download")
+                        .setContentText("Downloaded");

```

Comment:
title and text should be the same for both. I'd suggest "Download successful" as title and "Click to open "+filename, if there is an pendingintent to open the file on click (didnt check)